### PR TITLE
[homekit] simplify HomeKit console commands

### PR DESCRIPTION
**Pull request by yfre**
simplify HomeKit console commands:
- "list" instead of "listAccessories"
- "show" instead "printAccessory"

the commands are new and were introduce with recent HomeKit PRs so that we dont need legacy support.

### CODE REVIEW

File: bundles/org.openhab.io.homekit/README.md [hunk 1]
```diff
@@ -402,7 +402,7 @@ openhab> log:tail io.github.hapjava
 
 ## Console commands
 
-`smarthome:homekit listAccessories` - list all HomeKit accessory currently advertised to the HomeKit clients. the commands list the ID and name of accessories. 
+`smarthome:homekit list` - list all HomeKit accessory currently advertised to the HomeKit clients.
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/README.md:405]: Rather `accessories` ?

File: bundles/org.openhab.io.homekit/README.md [hunk 2]
```diff
@@ -402,7 +402,7 @@ openhab> log:tail io.github.hapjava
 
 ## Console commands
 
-`smarthome:homekit listAccessories` - list all HomeKit accessory currently advertised to the HomeKit clients. the commands list the ID and name of accessories. 
+`smarthome:homekit list` - list all HomeKit accessory currently advertised to the HomeKit clients.  
 
-`smarthome:homekit printAccessory <accessory_id>` - print additional details like list of characteristics for give accessory.
+`smarthome:homekit show <accessory_id | name>` - print additional details of the accessories which matche provided IDs or name.
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/README.md:407]: match

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java [hunk 3]
```diff
@@ -68,25 +68,27 @@ public void execute(String[] args, Console console) {
                     break;
                 case SUBCMD_PRINT_ACCESSORY:
                     if (args.length > 1) {
-                        Integer id = Integer.valueOf(args[1]);
-                        printAccessory(id, console);
+                        printAccessory(args[1], console);
                     } else {
-                        console.println("accessory id is required as an argument");
+                        console.println("accessory id or name are required as an argument");
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java:73]: is required

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java [hunk 4]
```diff
@@ -134,10 +136,11 @@ private void listAccessories(Console console) {
         });
     }
 
-    private void printAccessory(Integer accessory_id, Console console) {
+    private void printAccessory(String id, Console console) {
         homekit.getAccessories().forEach(v -> {
             try {
-                if (v.getId() == accessory_id) {
+                if (("" + v.getId()).contains(id) || ((v.getName().get() != null)
+                        && (v.getName().get().toUpperCase().contains(id.toUpperCase())))) {
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java:143]: This is no more a full matching but now a partial matching. Maybe this has to be mentioned in the usage ?

File: bundles/org.openhab.io.homekit/README.md (hunk 2 above)
Comment (yfre) [reply bundles/org.openhab.io.homekit/README.md]: oh. that many typos in just few changes sentences :( 
thank you the review. typos fixed.

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java [hunk 5]
```diff
@@ -68,25 +68,27 @@ public void execute(String[] args, Console console) {
...
         }
     }
 
     @Override
     public List<String> getUsages() {
         return Arrays.asList(buildCommandUsage(SUBCMD_LIST_ACCESSORIES, "list all HomeKit accessories"),
-                buildCommandUsage(SUBCMD_PRINT_ACCESSORY + " <accessory id>", "print accessorty details"),
-                buildCommandUsage(SUBCMD_CLEAR_PAIRINGS, "removes all pairings with HomeKit clients"),
+                buildCommandUsage(SUBCMD_PRINT_ACCESSORY + " <accessory id | accessory name>",
+                        "search for accessories and show details. "),
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java:90]: Please mention the partial matching here too.

Comment (yfre) [reply bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java]: good point. copied the description from README.

**Comment by J-N-K** [comment]
Since we released 2.5.5 with the "old" commands, I would prefer to keep them.  You can mark them `@Deprecated` and we can remove them for 3.0

**Comment by yfre** [comment]
> Since we released 2.5.5 with the "old" commands, I would prefer to keep them. You can mark them `@Deprecated` and we can remove them for 3.0

agree. i was hoping to get this PR into 2.5.5 but it is also no effort to add support for "old" commands. i have just commited the fix for it.

no idea what i was thinking by introducing "listAccessories" as the command. i have regretted it already on second time i had to type it. probably i was inspired by "clearPairings". anyway, both long commands and short commands are supported now.

**Comment by J-N-K** [comment]
I would suggest to emove the old commands from the documentation. That way nobody will use them if they look at the newest documenation abd probably only few people will be affected at all.

**Comment by yfre** [comment]
> I would suggest to emove the old commands from the documentation. That way nobody will use them if they look at the newest documenation abd probably only few people will be affected at all.

you are right, probably not that many people will missed them in the documentation. removed

**Pull request merged**